straxen/plugins/daqreader.py

```diff
@@ -259,9 +279,84 @@ def compute(self, chunk_i):
             # Ensure the offset is a whole digitizer sample
             records["time"] += self.dt * (self.t0 // self.dt)
 
-        result = self.chunk(
-            start=self.t0 + break_pre,
-            end=self.t0 + break_post,
-            data=records)
-        print(f"Read chunk {chunk_i:04d}, {result} from DAQ")
+        # Split records by channel
+        result_arrays = split_channel_ranges(
+            records,
+            np.asarray(list(self.config['channel_map'].values())))
+        del records
+
+        # Convert to strax chunks
+        result = dict()
+        for i, subd in enumerate(self.config['channel_map']):
+
+            # Ignore data from the 'blank' channels, corresponding to
+            # channels that have nothing connected
+            if subd.endswith('blank'):
+                continue
+
+            result_name = 'raw_records'
+            if subd != 'tpc':
+                result_name += '_' + subd
+            result[result_name] = self.chunk(
+                start=self.t0 + break_pre,
+                end=self.t0 + break_post,
+                data=result_arrays[i],
+                data_type=result_name)
+
+        print(f"Read chunk {chunk_i:06d} from DAQ")
+        for r in result.values():
+            print(f"\t{r}")
         return result
+
+
+@export
+class Fake1TDAQReader(DAQReader):
+    provides = (
+        'raw_records',
+        'raw_records_diagnostic',
```

Comment:
@JelleAalbers I think this does not work with the xnt_common_config as above since this will result in a key error.

Reply:
True: the Fake1TDAQReader is only meant to be run on XENON1T data through the fake_daq. XENONnT can use DAQReader.

Reply:
Thanks, I did a register all instead of register for daqreader rather than for DAQReader